backend/bots/examples/satoshi_dice/impl/src/updates/handle_direct_message.rs

```diff
@@ -0,0 +1,98 @@
+use crate::model::pending_actions_queue::{Action, TransferCkbtc};
+use crate::model::user_map::DiceRoll;
+use crate::{mutate_state, RuntimeState, MAX_SATS_PER_ROLL};
+use canister_api_macros::update_msgpack;
+use canister_tracing_macros::trace;
+use rand::RngCore;
+use satoshi_dice_canister::handle_direct_message::*;
+use types::{BotMessage, Cryptocurrency, MessageContent, TextContent, UserId};
+use utils::time::MINUTE_IN_MS;
+
+#[update_msgpack]
+#[trace]
+fn handle_direct_message(args: Args) -> Response {
+    mutate_state(|state| handle_message(args, state))
+}
+
+fn handle_message(args: Args, state: &mut RuntimeState) -> Response {
+    let mut messages = Vec::new();
+    if let Some(sats) = extract_ckbtc_amount(args.content) {
+        let user_id: UserId = state.env.caller().into();
+        let now = state.env.now();
+
+        if sats > MAX_SATS_PER_ROLL {
+            messages.push(
+                "The limit per roll is 10k SATS, so this roll won't count, please wait a moment while I refund your ckBTC"
+                    .to_string(),
+            );
+            send_ckbtc_message(user_id, sats.saturating_sub(20), state);
```

Comment:
I think 20 should be 2 * FEE for clarity

Reply:
Done